### ISSUE TITLE
const for for-of doesn't cause an error

### ISSUE THREAD

**Issue by kuraga** [Microsoft/TypeScript#7303]
**TypeScript Version:** 

1.8.0-beta

**Code**

```ts
for (const v of [1,2,3]);
```

**Expected behavior:**

A compiler error: re-assignment of `const`-variable in loop.

**Actual behavior:** 

Compiled successfully.

**Comment by DanielRosenwasser** [comment]
This is what's supposed to happen. A new environment is created per iteration with a bound variable named `v`. See https://tc39.github.io/ecma262/#sec-runtime-semantics-forin-div-ofbodyevaluation-lhs-stmt-iterator-lhskind-labelset.

**Comment by DanielRosenwasser** [comment]
(By the way, TypeScript 1.8 proper was released, you should totally update!)
